src/ucp/am/eager.inl

```diff
@@ -111,23 +111,26 @@ static ucs_status_t ucp_proto_request_eager_am_zcopy_init(
 }
 
 static void ucp_proto_eager_am_zcopy_add_footer(
-        ucp_request_t *req, size_t offset, ucp_lane_index_t lane,
+        ucp_request_t *req, size_t offset, ucp_rsc_index_t md_index,
         uct_iov_t *iov, size_t *iovcnt, size_t footer_size)
 {
     ucp_mem_desc_t *reg_desc = req->send.msg_proto.am.reg_desc;
-    ucp_ep_h ep              = req->send.ep;
     void *buffer;
-    ucp_md_index_t md_idx;
+    uct_mem_h mem_h;
 
     if (footer_size == 0) {
         return;
     }
 
     buffer = UCS_PTR_BYTE_OFFSET(reg_desc + 1, offset);
-    md_idx = ucp_ep_md_index(ep, lane);
 
-    ucp_add_uct_iov_elem(iov, buffer, footer_size, reg_desc->memh->uct[md_idx],
-                         iovcnt);
+    if (md_index == UCP_NULL_RESOURCE) {
```

Comment:
We use the same approach when we add iov with payload,
E.g. `ucp_datatype_iter_next_iov` calls `ucp_datatype_iter_uct_memh`, where we have:
```
static UCS_F_ALWAYS_INLINE uct_mem_h
ucp_datatype_iter_uct_memh(const ucp_mem_h memh, ucp_rsc_index_t memh_index)
{
    if (memh_index == UCP_NULL_RESOURCE) {
        return UCT_MEM_HANDLE_NULL;
    }

    ucs_assertv((UCS_BIT(memh_index) & memh->md_map) ||
                ucp_memh_is_zero_length(memh),
                "memh_index=%d md_map=0x%" PRIx64, memh_index, memh->md_map);
    return memh->uct[memh_index];
}
```

Reply:
ic, I guess it's because of https://github.com/openucx/ucx/blob/5edd29ba4fb46c85bfee9ef983946aaf7b4b22a5/src/ucp/proto/proto_common.c#L40 (which is not really needed). so can keep it for now.